src/openrct2/ride/Ride.cpp

```diff
@@ -5476,114 +5476,77 @@ sint32 ride_is_valid_for_open(sint32 rideIndex, sint32 goingToBeOpen, sint32 isA
     return 1;
 }
 
+/**
+ * Given a track element of the ride, find the start of the track.
+ * It has to do this as a backwards loop in case this is an incomplete track.
+ */
+void ride_get_start_of_track(CoordsXYE * output)
+{
+    track_begin_end trackBeginEnd;
+    CoordsXYE trackElement = *output;
+    if (track_block_get_previous(trackElement.x, trackElement.y, trackElement.element, &trackBeginEnd)) {
+        rct_tile_element* initial_map = trackElement.element;
+        do {
+            CoordsXYE lastGood = {
+                /* .x = */ trackBeginEnd.begin_x,
+                /* .y = */ trackBeginEnd.begin_y,
+                /* .element = */ trackBeginEnd.begin_element
+            };
+
+            if (!track_block_get_previous(trackBeginEnd.end_x, trackBeginEnd.end_y, trackBeginEnd.begin_element, &trackBeginEnd)) {
+                trackElement = lastGood;
+                break;
+            }
+        } while (initial_map != trackBeginEnd.begin_element);
+    }
+    output = &trackElement;
+}
+
 /**
  *
  *  rct2: 0x006CB7FB
  */
 sint32 ride_get_refund_price(sint32 ride_id)
 {
-    uint8 oldpaused = gGamePaused;
-    gGamePaused = 0;
-    money32 refundPrice = 0;
-
-    tile_element_iterator it;
-
-    tile_element_iterator_begin(&it);
-    while (tile_element_iterator_next(&it)) {
-        if (it.element->GetType() != TILE_ELEMENT_TYPE_TRACK)
-            continue;
+    CoordsXYE trackElement;
+    money32 addedcost, cost = 0;
 
-        if (track_element_get_ride_index(it.element) != ride_id)
-            continue;
+    if (!ride_try_get_origin_element(ride_id, &trackElement)) {
+        gGameCommandErrorText = STR_TRACK_TOO_LARGE_OR_TOO_MUCH_SCENERY;
+        return MONEY32_UNDEFINED;
+    }
 
-        sint32 x = it.x * 32, y = it.y * 32;
-        sint32 z = it.element->base_height * 8;
+    // Find the start in case it is not a complete circuit
+    ride_get_start_of_track(&trackElement);
 
-        uint8 rotation = tile_element_get_direction(it.element);
-        uint8 type = track_element_get_type(it.element);
+    uint8 direction = tile_element_get_direction(trackElement.element);
 
-        if (type != TRACK_ELEM_INVERTED_90_DEG_UP_TO_FLAT_QUARTER_LOOP){
```

Comment:
This check appears to have been lost, perhaps in a rebase at some point?

Reply:
This check is still there, but [here](https://github.com/OpenRCT2/OpenRCT2/blob/77b09b3daffb7c0ba972bc5b71e1b2af5284b1ef/src/openrct2/actions/RideDemolishAction.hpp#L271). The old ride_get_refund_price function was no longer used (and had a wrong name, should have been named DemolishTracks), so I completely rewrote that function to do what it says and used it.
TLDR: the removed code in ride_get_refund_price has already been moved.